test/enhance/test_adjust.py

```diff
@@ -42,38 +43,37 @@ def test_saturation_one_batch(self, device, dtype):
                                [.5, .5]],
 
                               [[.25, .25],
-                               [.25, .25]]]], device=device)  # 2x3x2x2
+                               [.25, .25]]]], device=device, dtype=dtype)  # 2x3x2x2
 
         expected = data
         f = kornia.enhance.AdjustSaturation(torch.ones(2))
         assert_allclose(f(data), expected)
 
-    def test_gradcheck(self, device):
+    def test_gradcheck(self, device, dtype):
         batch_size, channels, height, width = 2, 3, 4, 5
-        img = torch.rand(batch_size, channels, height, width)
+        img = torch.rand(batch_size, channels, height, width, device=device, dtype=dtype)
         img = tensor_to_gradcheck_var(img)  # to var
         assert gradcheck(kornia.adjust_saturation, (img, 2.),
                          raise_exception=True)
 
 
 class TestAdjustHue:
-    def test_hue_one(self, device):
+    def test_hue_one(self, device, dtype):
```

Comment:
Just wandering if to have a BaseTester for all modulized functions to perform some routined tests as well. /cc @FGeri

Reply:
we can have a seconf iteration for BaseTester and automate a lot of things